### ISSUE TITLE
Mafia III [GOG] freezes at splash screen

### ISSUE THREAD

**Issue by dreamhunt** [doitsujin/dxvk#298]
Hello and thank you for your GREAT work :)
Mafia III [GOG] freezes at splash screen with DXVK. I tried it with 0.41 and 0.42. The game starts normally with Wine and Wine-staging but is with messed graphics which is very dark. With DXVK it freezes at splash screen. With apitrace the game again freezes at splash screen.

### Software information
Mafia III [GOG]
Arch Linux 86_64

### System information
- GPU: Intel 3770
- Driver: 390.48, Nvidia GTX 1080Ti
- Wine version: 3.5 Staging/ 3.5 vanilla
- DXVK version: 0.42

### Apitrace file(s)
https://pastebin.com/WQiJUtRJ

### Log files
- d3d11.log:
https://pastebin.com/Pv1UMUJA
- dxgi.log: 
https://pastebin.com/Af1V1zPC
Wine log:
https://pastebin.com/HXBx0p7X

**Comment by doitsujin** [comment]
This game is known to require Stream Output, which just isn't implemented yet. Based on your logs it's obvious that the game tries and fails to create a geometry shader with stream output - that doesn't mean that it causes the crash, but it doesn't make sense to dig into that further before the issue is fixed, especially since I cannot reproduce it. See #135

**Comment by dreamhunt** [comment]
But with normal Wine the game runs beyond the splash screen and I was hoping that with DXVK it may be with normal graphics, Sorry for the inconvenience.

**Comment by doitsujin** [comment]
Wine implements Stream Output, DXVK doesn't.

**Comment by dreamhunt** [comment]
Ok we'll wait for Stream Output implementation than. Thank you :)

**Comment by vanyasem** [comment]
Mafia III works great if you compile DXVK with fake output support

![2018-08-22-030556_1931x1130_scrot](https://user-images.githubusercontent.com/13350458/44465129-57182100-a625-11e8-823e-aa759ba9fdc3.png)

**Comment by dreamhunt** [comment]
That's GREAT news!!! Would you like to give us precompiled libraries with fake output support? Please :)

**Comment by dreamhunt** [comment]
Now I tried it with dxvk.conf but it doesn't work. The game still freezes at the splash screen. Please, give me two x64 libraries of yours. Thank you in advance.

**Comment by vanyasem** [comment]
Arch packages: https://drive.google.com/drive/folders/1bCHCtJHAAss2NwGNcgwdEx2WHdE5DT83?usp=sharing

**Comment by dreamhunt** [comment]
Thank you very much! I replaced the libraries with those yuo gave me but the game still freezes at startup. I tried it with new, clean prefix to no avail. Did you installed some other staff? I try GOG version. Is your version different?
Thank you once again :)

**Comment by vanyasem** [comment]
I am using the Steam version, but the GOG version is also known to work. What I've done (according to the Lutris script) is that I renamed `mafia3.exe` to `launcher.exe` (and renamed `launcher.exe` to `mafia3.exe` accordingly). The launcher requires DotNet 4, and isn't supported by Wine (it crashes even if you install DotNet/wine-mono)

**Comment by vanyasem** [comment]
Oh and also the most important part, you should set the game to Windowed mode (fullscreen is known to freeze at startup). For that open `$WINEPREFIX/drive_c/users/vanyasem/Local Settings/Application Data/2K Games/Mafia III/Saves/videoconfig.cfg` and paste `-6 0 1920 1080 0 0 0 0` (if the `Saves` folder doesn't exist, create it)

**Comment by mozo78** [comment]
Yes, the game runs now! The fulscreen part was the culprit. Thank you very much again. Finally I'll play it!

**Comment by dreamhunt** [comment]
Yes, the game runs now! The fulscreen part was the culprit. Thank you very much again. Finally I'll play it!

**Comment by mozo78** [comment]
Hello again :)
I actually found a way to run the game in full screen and I want to share it with you. I hope it will be benefical for somebody.
If you run on lower resolution than defined in the videoconfig.cfg, the game will run in fullscreen without problems. For example, if your resolution is 1280x800 and you set in videoconfig.cfg the following:
`0 0 1920 1080 1 0 0 1`
the game will run fine in fullscreen at 1280x800 resolution.
I hope this will be useful.

@vanyasem,
You can set the fps to 60 from game options. I'm with 3770 and GTX1080Ti and the game runs with stable 55-60 FPS :)

**Comment by vanyasem** [comment]
My framerate is set to unlimited. I run the game in 4k, that's why it's slow :P @mozo78

**Comment by mozo78** [comment]
Ah, I see :)

**Comment by brenthuisman** [comment]
@vanyasem Can your dxvk builds be used on other distros?

**Comment by vanyasem** [comment]
@brenthuisman you're welcome to try. I only run Arch

**Comment by mozo78** [comment]
Yes, the builds can be used on other distributions. I tried them on Arch and Mint 19 and they work without problems.

**Comment by mozo78** [comment]
I just completed the game:
![1](https://user-images.githubusercontent.com/38666396/45588777-127d5e00-b923-11e8-80be-d12f667d23f1.png)
It was tough task though. The game has random crashes and I had to replay some missions several times. In some area the game suffers from HEAVY frame drop around 1-2 FPS until you go out of these areas. Often if you return to that area, the game again crashes. Anyway I'm a super, hyper stubborn person and completed it. Thank you VERY much doitsujin, vanyasem!!!

**Comment by mozo78** [comment]
Hello again. Even with Stream Output Support the game has some issues. I found two. The first one is negative like colors in some areas:
![mafia iii](https://user-images.githubusercontent.com/38666396/47615696-e83ec480-daba-11e8-96d5-119cb77dc6cc.png)
And the second is the heavy FPS drop in some areas. It is below 1 FPS and it's very hard to avoid such areas. I can't test with vanilla Wine because the game just freezes. The first such area is when Lincon have to call his boss after the boat escape from the bank. Once the call has ended and the game suffers from severe framedrop. If you are patient enough, you'll walk away frome the phone area and the game than runs normal again. There are many such areas further in the game.

**Comment by doitsujin** [comment]
Can anyone confirm this on AMD?

**Comment by mozo78** [comment]
The colors are greysh and negative like. The must be colorful not almost black and white. Here you can see the same location from this quest:
https://youtu.be/Ps2b8vbHJa0?t=258

**Comment by doitsujin** [comment]
I don't see the "almost black and white" part in your screenshot. Maybe it's too bright, but that might as well be because it's a different time of day? Don't know.

If you want me to debug it, please record an apitrace.

**Comment by mozo78** [comment]
The severe framedrop is the biggest issue. Don't bother with the negative effect :) Let someone say if it occurs on AMD too.

**Comment by doitsujin** [comment]
There is no "negative effect" and if it's a DXVK bug, i very much **do** care. More so than about frame drops because those probably cannot be fixed anyway.

**Comment by mozo78** [comment]
Unfortunately I have alredy completed the game and today I just tried it Stream Output to see how it behaves. After that I deleted the Wine prefix and the saves are gone. I don't want to play the begining for a third time to reach thus ques again. And this problem isn't a game braker but the problem with the severe frame drop does. One has to be with VERY stong nerves to play FPS with 0.5 FPS especially when the AI shoot at him.
